pkg/proxy/command.go

```diff
@@ -263,6 +283,119 @@ func ExtractCommandAction(ctx context.Context, command string, params RequestPar
 		return func(v *vehicle.Vehicle) error { return v.Unlock(ctx) }, nil
 	case "erase_user_data":
 		return func(v *vehicle.Vehicle) error { return v.EraseGuestData(ctx) }, nil
+	case "add_charge_schedule":
+		lat, err := params.getNumber("lat", true)
+		if err != nil {
+			return nil, err
+		}
+		lon, err := params.getNumber("lon", true)
+		if err != nil {
+			return nil, err
+		}
+		startTime, err := params.getNumber("start_time", false)
+		if err != nil {
+			return nil, err
+		}
+		startEnabled, err := params.getBool("start_enabled", true)
+		if err != nil {
+			return nil, err
+		}
+		endTime, err := params.getNumber("end_time", false)
+		if err != nil {
+			return nil, err
+		}
+		endEnabled, err := params.getBool("end_enabled", true)
+		if err != nil {
+			return nil, err
+		}
+		daysOfWeek, err := params.getDays("days_of_week", true)
+		if err != nil {
+			return nil, err
+		}
+		id, err := params.getNumber("id", false)
+		if err != nil {
+			return nil, err
+		}
+		idUint64 := uint64(id)
+		if id == 0 {
+			idUint64 = uint64(time.Now().Unix())
+		}
+		enabled, err := params.getBool("enabled", true)
+		if err != nil {
+			return nil, err
+		}
+		oneTime, err := params.getBool("one_time", false)
+		if err != nil {
+			return nil, err
+		}
+		schedule := vehicle.ChargeSchedule{
+			DaysOfWeek:   daysOfWeek,
+			Latitude:     float32(lat),
+			Longitude:    float32(lon),
+			Id:           idUint64,
+			StartTime:    int32(startTime),
+			EndTime:      int32(endTime),
+			StartEnabled: startEnabled,
+			EndEnabled:   endEnabled,
+			Enabled:      enabled,
+			OneTime:      oneTime,
+		}
+		return func(v *vehicle.Vehicle) error { return v.AddChargeSchedule(ctx, &schedule) }, nil
+	case "add_precondition_schedule":
+		lat, err := params.getNumber("lat", true)
+		if err != nil {
+			return nil, err
+		}
+		lon, err := params.getNumber("lon", true)
+		if err != nil {
+			return nil, err
+		}
+		preconditionTime, err := params.getNumber("precondition_time", true)
+		if err != nil {
+			return nil, err
+		}
+		oneTime, err := params.getBool("one_time", false)
+		if err != nil {
+			return nil, err
+		}
+		daysOfWeek, err := params.getDays("days_of_week", true)
+		if err != nil {
+			return nil, err
+		}
+		id, err := params.getNumber("id", false)
```

Comment:
should we generate the id?

Reply:
read down a few lines. If id is passed, we use that. Otherwise, generate id using current time.